openedx/features/enterprise_support/api.py

```diff
@@ -233,6 +234,32 @@ def enterprise_enabled():
     return 'enterprise' in settings.INSTALLED_APPS and getattr(settings, 'ENABLE_ENTERPRISE_INTEGRATION', True)
 
 
+def enterprise_customer_for_request(request, tpa_hint=None):
```

Comment:
Okay! I understand where you're coming from, both here and above. However, I think we're moving into the category of "existing tech debt". I don't disagree with you on any particular point, but what you're suggesting _is_ a substantial alteration of how things have been done so far - which means that we ought to sit down and have a substantive architecture discussion, as well as an epic or story to move the overall implementation to meet your expectations - because the architecture you seem to be describing is _absolutely_ not what we have at the moment. 

We've been operating under the believe that the `enterprise_support` module would be the sole connection point to the Enterprise app, with any methods inside it that need to talk to the Enterprise data backing being switched out to use HTTP APIs.